calendar-tool/tool/src/java/org/sakaiproject/calendar/entityproviders/CalendarEventEntityProvider.java

```diff
@@ -94,95 +102,72 @@ public boolean entityExists(final String id) {
 
 	/**
 	 * site/siteId
+	 *
+	 * Optional firstDate and lastDate query params in ISO-8601 date format, YYYY-MM-DD
 	 */
 	@EntityCustomAction(action = "site", viewKey = EntityView.VIEW_LIST)
-	public List<CalendarEventSummary> getCalendarEventsForSite(EntityView view) {
-		List<CalendarEventSummary> r = new ArrayList<CalendarEventSummary>();
+	public List<CalendarEventSummary> getCalendarEventsForSite(final EntityView view, final Map<String, Object> params) {
+
+		final List<CalendarEventSummary> rv = new ArrayList<CalendarEventSummary>();
 
 		// get siteId
-		String siteId = view.getPathSegment(2);
+		final String siteId = view.getPathSegment(2);
 
 		// check siteId supplied
 		if (StringUtils.isBlank(siteId)) {
 			throw new IllegalArgumentException(
-					"siteId must be set in order to get the news feeds for a site, via the URL /news/site/siteId");
+					"siteId must be set in order to get the calendar feeds for a site, via the URL /calendar/site/siteId");
 		}
 
-		// user being logged in and having access to the site is handled in the
-		// API
-		Calendar cal;
 		try {
-			cal = calendarService.getCalendar(String.format(
-					"/calendar/calendar/%s/main", siteId));
+			config.load(getClass().getResourceAsStream("/org/sakaiproject/calendar/tool/calendar.config"));
```

Comment:
You shouldn't do initialisation on a request like this, while this shouldn't have threading issues it can lead to them down the line, it's much cleaner todo it in an `init()`.